dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops

```diff
@@ -238,30 +238,40 @@ UNKNOWN_PLATFORM_VALUE_ENV_VAR=CWWKF0048E: The {0} value of the PREFERRED_PLATFO
 
-# CWWKF0051I  {0} = the versionless feature: {1} = the feature that was resolved from the versionless feature
-VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE=CWWKF0051I: The {0} versionless feature resolved to {1}
-VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.explanation=The versioned feature resolved from the versionless feature.
+# CWWKF0051I  {0} = the versionless features: {1} = the features that were resolved from the versionless features
+VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE=CWWKF0051I: The {0} versionless features resolved to {1}
+VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.explanation=The versioned features resolved from the versionless features.
 VERSIONLESS_FEATURE_RESOLVED_TO_FEATURE.useraction=No action is required.
 
-# CWWKF0052E  {0} = the unknown platform element
-UNKNOWN_PLATFORM_ELEMENT=CWWKF0052E: The {0} platform element is not a known platform
+# CWWKF0052E  {0} = the unknown platform
+UNKNOWN_PLATFORM_ELEMENT=CWWKF0052E: The {0} platform is not valid
 UNKNOWN_PLATFORM_ELEMENT.explanation=The specified platform element is not a known platform. The known platforms are a version of javaee, jakartaee, or microProfile
 UNKNOWN_PLATFORM_ELEMENT.useraction=Make sure the configured platform is a known platform.
 
-# CWWKF0053E  {0} = the unknown platform element
-UNKNOWN_PLATFORM_VERSION=CWWKF0053E: The {0} platform element is not a known platform version
-UNKNOWN_PLATFORM_VERSION.explanation=The platform is known, but the version of the platform does not exist.
-UNKNOWN_PLATFORM_VERSION.useraction=Specify a version of the platform that exists.
+# CWWKF0053I  {0} = the resolved platforms
+RESOLVED_PLATFORM=CWWKF0053I: Feature resolution selected the {0} platforms.
+RESOLVED_PLATFORM.explanation=The platform versions were selected to enable resolution of versionless features. The platform versions were obtained from platform elements, from versioned features, or from an environment variable.
+RESOLVED_PLATFORM.useraction=No action is required.
+
+# CWWKF0054E  {0} = the versionless feature that cannot be resolved: {1} = the resolved platform
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM=CWWKF0054E: The {0} versionless feature does not have a version belonging to the {1} platform.
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.explanation=No version of the feature is available for the platform version. Feature resolution was completed, but did not select a version of the feature.
+INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM.useraction=Either, remove the versionless feature, or specify a different platform version, or use a versioned feature.
+
+# CWWKF0055E 
+NO_RESOLVED_PLATFORM=CWWKF0055E: Versionless features are being used but there is no platform.
+NO_RESOLVED_PLATFORM.explanation=A platform must be determined before versionless features can be resolved. The platform can be determined from a platform element, from a versioned feature, or from an environment variable.
```

Comment:
```suggestion
NO_RESOLVED_PLATFORM.explanation=A platform must be specified before versionless features can be resolved. The platform can be specified in a platform element, from a versioned feature, or in an environment variable.
```

Reply:
this is a bit confusing- how is the platform determined from a versioned feature? Do you even need a platform if youre using a versioned feature?

Reply:
The feature files contain metadata that tells us what platforms they are a part of, and we use that as one of the ways determine a platform. You do not need a platform if you are just using versioned features, a platform is only needed when using versionless features.